packages/block-editor/src/components/block-list/use-block-props/use-selected-block-event-handlers.js

```diff
@@ -57,6 +57,7 @@ export function useEventHandlers( { clientId, isSelected } ) {
 				event.preventDefault();
 
 				if ( keyCode === ENTER ) {
+					target.blur();
```

Comment:
Will adding this cause issues with any other blocks?

I haven't thought through all of the scenarios yet, but I'm wondering if there's any case in which we _wouldn't_ want to blur the target. By blurring it, we ensure that focus gets set to the newly created block. Any insights here would be appreciated!

Reply:
cc @ellatrix Do you foresee any problems with this approach?

Reply:
@michalczaplinski @ellatrix As an alternative, I've [roughly extracted](https://github.com/WordPress/gutenberg/pull/58958/commits/31b622cf9bf30b682021f9823a70a0c12082d5e6) the RichText's `shouldDisableEditing` logic into a separate hook. If we want to go this route, we should perhaps rename the hook from `useShouldDisableEditing` to something more explanatory.

I'm not sure this is necessary, but it would probably be the safest approach. What do you think?

Reply:
> Will adding this cause issues with any other blocks?

What is the purpose of using `target.blur()`? In which cases it is not working without it?

From my tests, I can see that without that, everything seems to work fine. However, when `target.blur()` is added, if I follow these steps it doesn't work as expected:
1. Reload the page.
2. Select in the block list the paragraph with bindings.
3. Click enter.
4. It doesn't add a new block.

Reply:
> It doesn't add a new block.

Hmm, I'm not seeing this. But check out [this comment](https://github.com/WordPress/gutenberg/pull/58958#discussion_r1498266352) for additional ideas regarding the blur.